staging/src/k8s.io/apiserver/pkg/storage/cacher/watch_cache.go

```diff
@@ -303,8 +303,10 @@ func (w *watchCache) waitUntilFreshAndBlock(resourceVersion uint64, trace *utilt
 	}
 	for w.resourceVersion < resourceVersion {
 		if w.clock.Since(startTime) >= blockTimeout {
-			// Timeout with retry after 1 second.
-			return errors.NewTimeoutError(fmt.Sprintf("Too large resource version: %v, current: %v", resourceVersion, w.resourceVersion), 1)
+			// Timeout with retry after 'blockTimeout' seconds.
+			return storage.NewResourceVersionTooLargeError(int64(resourceVersion),
+				fmt.Sprintf("Too large resource version: %v, current: %v. Timeout after retry after %v",
+					resourceVersion, w.resourceVersion, blockTimeout))
```

Comment:
The previous error resulted in a Retry-After header presented to the API client. Are we intentionally dropping that behavior?

Reply:
Good eye. It’s still desired. I’ll fix and add a test.

Reply:
Fixed this by reverting back to using `NewTimeoutError` but adding a `ErrStatus.Details.Causes` item with a "Too large resource version" message to make it easy to check for these errors when expected in tests.